tsconfig.base.json

```diff
@@ -10,46 +10,89 @@
-      "@sagebionetworks/openchallenges/login": ["libs/openchallenges/login/src/index.ts"],
-      "@sagebionetworks/openchallenges/not-found": ["libs/openchallenges/not-found/src/index.ts"],
+      "@sagebionetworks/openchallenges/config": [
+        "libs/openchallenges/config/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/home": [
+        "libs/openchallenges/home/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/login": [
+        "libs/openchallenges/login/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/not-found": [
+        "libs/openchallenges/not-found/src/index.ts"
+      ],
       "@sagebionetworks/openchallenges/org-profile": [
         "libs/openchallenges/org-profile/src/index.ts"
       ],
-      "@sagebionetworks/openchallenges/org-search": ["libs/openchallenges/org-search/src/index.ts"],
-      "@sagebionetworks/openchallenges/pages": ["libs/openchallenges/pages/src/index.ts"],
-      "@sagebionetworks/openchallenges/signup": ["libs/openchallenges/signup/src/index.ts"],
-      "@sagebionetworks/openchallenges/styles": ["libs/openchallenges/styles/src/index.ts"],
-      "@sagebionetworks/openchallenges/team": ["libs/openchallenges/team/src/index.ts"],
-      "@sagebionetworks/openchallenges/themes": ["libs/openchallenges/themes/src/index.ts"],
-      "@sagebionetworks/openchallenges/ui": ["libs/openchallenges/ui/src/index.ts"],
+      "@sagebionetworks/openchallenges/org-search": [
+        "libs/openchallenges/org-search/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/pages": [
+        "libs/openchallenges/pages/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/signup": [
+        "libs/openchallenges/signup/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/styles": [
+        "libs/openchallenges/styles/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/team": [
+        "libs/openchallenges/team/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/themes": [
+        "libs/openchallenges/themes/src/index.ts"
+      ],
+      "@sagebionetworks/openchallenges/ui": [
+        "libs/openchallenges/ui/src/index.ts"
+      ],
       "@sagebionetworks/openchallenges/user-profile": [
         "libs/openchallenges/user-profile/src/index.ts"
       ],
-      "@sagebionetworks/openchallenges/util": ["libs/openchallenges/util/src/index.ts"],
-      "@sagebionetworks/shared/util": ["libs/shared/typescript/util/src/index.ts"],
+      "@sagebionetworks/openchallenges/util": [
+        "libs/openchallenges/util/src/index.ts"
+      ],
+      "@sagebionetworks/shared/util": [
+        "libs/shared/typescript/util/src/index.ts"
+      ],
       "@sagebionetworks/shared/web-components": [
         "libs/shared/typescript/web-components/src/index.ts"
+      ],
+      "@sagebionetworks/model-ad/ui": [
+        "libs/model-ad/ui/src/index.ts"
       ]
```

Comment:
I'm wondering if the proposed definitions in `tsconfig.base.json` would be useful for non-TypeScript libraries (styles and themes are SCSS files). I did try to add the definitions to this files and then tried to use one:

`tsconfig.base.json`:

```
      "@sagebionetworks/model-ad/styles": [
        "libs/model-ad/styles/src/index"
      ]
```

`apps/model-ad/app/src/styles.scss`:

```
/* You can add global styles to this file, and also import other style files */
// @use 'libs/model-ad/styles/src/index';
@use '@sagebionetworks/model-ad/styles';

@use 'app-theme';
```

This would have been nice if it worked as we would no longer had to rely on an absolute path. However, trying to start the app generate the following error (maybe the definition in `tsconfig.base.json` is incorrect):

```
./apps/model-ad/app/src/styles.scss - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
  ╷
3 │ @use '@sagebionetworks/model-ad/styles';
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  apps/model-ad/app/src/styles.scss 3:1  root stylesheet

./apps/model-ad/app/src/styles.scss?ngGlobalStyle - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Module build failed (from ./node_modules/@angular-devkit/build-angular/node_modules/sass-loader/dist/cjs.js):
Can't find stylesheet to import.
  ╷
3 │ @use '@sagebionetworks/model-ad/styles';
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  apps/model-ad/app/src/styles.scss 3:1  root stylesheet
```

I propose to leave the styles and themes libraries out of the tsconfig file until we come across a case where these definitions are useful.

Reply:
That makes sense to me! I'd noticed that the styles/themes openchallenges libraries are listed in tsconfig.base.json, so wasn't sure if the model-ad libraries needed to be added as well, but sounds good to wait until these definitions are useful!